### ISSUE TITLE
Feature: Ignore errors when keys do not exist on system.

### ISSUE THREAD

**Issue by mkristan** [sous-chefs/sysctl#110]
### Cookbook version
1.0.0

### Chef-client version
14.0.59

### Platform Details
Amazon Linux 2017.09

### Scenario:
Feature request for a way to define in the resource to ignore errors when a given key does not exist on a platform, this will make wrapper cookbooks easier to implement as there won't be a need to introduce complex guards in each resource.

### Steps to Reproduce:
Declare a sysctl paramter that does not exist on the system.  For example net.ipv4.tcp_syncookies.  See travis output at https://travis-ci.org/sous-chefs/sysctl/jobs/342817880 

### Expected Result:
Previous versions of the cookbook would silently fail if the key does not exist on a target system.  

### Actual Result:
The cookbook throws an exception because the sysctl -p command returns an error 255.  Full details below

    Mixlib::ShellOut::ShellCommandFailed
    ------------------------------------
    execute[sysctl -p] (/opt/kitchen/cache/cookbooks/sysctl/resources/param.rb line 62) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '255'

**Comment by damacus** [comment]
yeah this is super bloody annoying, and it's currently limited to travis. 

I actually think this is an OS configuration error rather than a sysctl issue. 

But if you want to add in an ignore_errors switch a PR would be welcomed! 👍

**Comment by mkristan** [comment]
I can look into a PR.

**Comment by mkristan** [comment]
I'm still looking into this.  This appears to be in the code a while back but seems to have gotten lost when we switched from attributes to resources.

**Comment by cpoma** [comment]
The "quick" fix in proc_hard is:
```ruby
node['sysctl']['params'].each do |param, value|
  sysctl_param param do
    key param
    value value
    only_if "sysctl -n -e #{param}"
    return [0,255]
  end
end
```
